### DISCUSSION TITLE
Deploying to Vercel (Error)

### DISCUSSION THREAD

**Discussion by viperfx** [quirrel-dev/quirrel#173] [Q&A] [answered]
I was trying to deploy to vercel and I have my build command set to the following

`yarn rw deploy vercel && yarn run quirrel ci`

I am getting the following error

```

15:41:40.024 | (node:2723) UnhandledPromiseRejectionWarning: FetchError: request to http://localhost:9181/queues/ failed, reason: connect ECONNREFUSED 127.0.0.1:9181
-- | --
15:41:40.024 | at ClientRequest.<anonymous> (/vercel/workpath0/node_modules/node-fetch/lib/index.js:1461:11)
15:41:40.024 | at ClientRequest.emit (events.js:315:20)
15:41:40.024 | at ClientRequest.EventEmitter.emit (domain.js:467:12)
15:41:40.024 | at Socket.socketErrorListener (_http_client.js:469:9)
15:41:40.024 | at Socket.emit (events.js:315:20)
15:41:40.024 | at Socket.EventEmitter.emit (domain.js:467:12)
15:41:40.025 | at emitErrorNT (internal/streams/destroy.js:106:8)
15:41:40.025 | at emitErrorCloseNT (internal/streams/destroy.js:74:3)
15:41:40.025 | at processTicksAndRejections (internal/process/task_queues.js:80:21)
15:41:40.025 | (Use `node --trace-warnings ...` to show where the warning was created)
15:41:40.028 | (node:2723) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
15:41:40.028 | (node:2723) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```

I should have the correct env variables set.

Any clue on why that's happening?

**Comment by Skn0tt** [comment, marked as answer]
Quirrel relies on the environment variables `CI`, `VERCEL` (if available) and `NODE_ENV` to figure out the endpoint it should talk against (see [`src/client/config.ts`](https://github.com/quirrel-dev/quirrel/blob/main/quirrel/src/client/config.ts)).

Did you change something with these variables?

**Reply by viperfx** [reply]
Ah - yes - I manually enable system variables as needed! I am at the 4KB limit with my application :)

**Reply by Skn0tt** [reply]
I see! So that's likely the reason.

**Reply by viperfx** [reply]
@Skn0tt hmm I don't see VERCEL and CI as system variables in their UI that can be added. Are these custom ones that we should add? 

If not they are most likely already in the system.

**Reply by viperfx** [reply]
Also just to make sure - QUIRREL_BASE_URL I should not be setting on our staging or production servers right? 

Unfortunately VERCEL currently only allows exposing all the variables, and I am hit with the 4kB limit. I am trying a work around by setting NODE_ENV to production

**Reply by Skn0tt** [reply]
`QUIRREL_BASE_URL` is intended for self-hosted instances, but could actually be the solution to your problem. Could you try setting it to `https://api.quirrel.dev`?

**Reply by viperfx** [reply]
So I set NODE_ENV manually and I got the following

```
17:02:27.106 | (node:2259) UnhandledPromiseRejectionWarning: Error: Please specify QUIRREL_BASE_URL.
```

Will try your solution

**Reply by viperfx** [reply]
I set the base url env to that value and re-deployed.

Got the same 

```
(node:2306) UnhandledPromiseRejectionWarning: FetchError: request to http://localhost:9181/queues/ failed, reason: connect ECONNREFUSED 127.0.0.1:9181
```

**Reply by viperfx** [reply]
@Skn0tt does other env variables need to be present?

**Reply by Skn0tt** [reply]
Oh, I mixed up `QUIRREL_BASE_URL` with `QUIRREL_URL`. Sorry, my bad 😬

`QUIRREL_URL` should be set to `https://api.quirrel.dev`. This is used to determine what Quirrel API to talk against. If it's set explicitely, it overrides any defaults inferred from `CI` or `VERCEL` or `NODE_ENV`.

`QUIRREL_BASE_URL`, on the other hand, is used to determine how the the Quirrel API can reach your application. You likely already had that set correctly.

**Reply by viperfx** [reply]
Is there a way we can cut this down? This application currently requires 

QUIRREL_ENCRYPTION_SECRET
QUIRREL_TOKEN
QUIRREL_BASE_URL
QUIRREL_URL

Would be great if this could be reduced

@Skn0tt

**Reply by viperfx** [reply]
Btw I set all of them and it deployed without blowing up! @Skn0tt

**Reply by Skn0tt** [reply]
> Btw I set all of them and it deployed without blowing up!

Nice!

> Is there a way we can cut this down?

No, sadly not. I'm sure there's a way to remove `QUIRREL_URL`, but the other ones are necessary.
